pkg/cli/clients/management.go

```diff
@@ -42,8 +42,13 @@ import (
 )
 
 type UCPApplicationsManagementClient struct {
-	RootScope     string
-	ClientOptions *arm.ClientOptions
+	RootScope                        string
+	ClientOptions                    *arm.ClientOptions
+	genericResourceClientFactory     func(scope string, resourceType string) (genericResourceClient, error)
+	applicationResourceClientFactory func(scope string) (applicationResourceClient, error)
+	environmentResourceClientFactory func(scope string) (environmentResourceClient, error)
+	resourceGroupClientFactory       func() (resourceGroupClient, error)
+	capture                          func(ctx context.Context, capture **http.Response) context.Context
```

Comment:
Please review this file in detail. Each of these functions have been updated to be more consistent (naming, structure, etc). 

Adding mocks was pretty complex.